src/v/redpanda/admin_server.cc

```diff
@@ -3090,7 +3090,7 @@ void admin_server::register_shadow_indexing_routes() {
           auto ntp = model::ntp(model::kafka_namespace, topic, partition);
           if (need_redirect_to_leader(ntp, _metadata_cache)) {
               vlog(logger.info, "Need to redirect bucket syncup request");
-              co_await redirect_to_leader(*request, ntp);
+              throw co_await redirect_to_leader(*request, ntp);
```

Comment:
nice catch. we should probably consider marking redirect_to_leader [[nodiscard]]